### DISCUSSION TITLE
Conventional Commits

### DISCUSSION THREAD

**Discussion by stritti** [remotemobprogramming/mob#391] [Q&A] [answered]
We use [Conventional Commits](https://www.conventionalcommits.org/) in our environment. any suggestions how to configure `mob` to work with this convention?

**Comment by hollesse** [comment]
Hey @stritti could you give me some more information about using conventional commits? I had a look at the website and if you are just using the convention using mob should be fine as you can enter the commit message when doing mob done. Or is the WIP commit message a problem, because you check that every commit is in the conventional commit spec?

**Reply by hollesse** [reply]
If you lint the commit messages please tell me with which tool you lint the commit messages such that I could check out the rules. For example which types are allowed, as the specification itself allows to add your own types.

**Reply by stritti** [reply]
I receive following error from our GitLab:

```sh
  git fetch origin --prune
  git merge FETCH_HEAD --ff-only
> starting new session from origin/develop
  git checkout -B mob/develop origin/develop
  git commit --allow-empty -m mob start [ci-skip] [ci skip] [skip ci]
ERROR git push --no-verify --set-upstream origin mob/develop:mob/develop
ERROR remote: GitLab: Commit message does not follow the pattern '^(Notes added by \'git notes add\')|((build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test|¯\\_\(ツ\)_\/¯)(\([\w\-\#]+\))?(!)?(: (.*\s*)*))|(Merge|Revert|Resolve [^\s].*)|(Initial commit$)'     
ERROR To gitlab.***.de:/***.git
ERROR ! [remote rejected] mob/develop -> mob/develop (pre-receive hook declined)
```

**Reply by hollesse** [reply, marked as answer]
Hey @stritti,
I just had a look. There is an easy way to get it working :) 
First of all finish all mob branches in your project. 
After that I would recommend creating a `.mob` file in your project root. There you could add the following code, to adjust the mob start and mob next commit message:
```
MOB_WIP_COMMIT_MESSAGE="¯\_(ツ)_/¯: mob next [ci-skip] [ci skip] [skip ci]"
MOB_START_COMMIT_MESSAGE="¯\_(ツ)_/¯: mob start [ci-skip] [ci skip] [skip ci]"
```

This code does match the regex and as the WIP commits will be squashed by doing mob done this message with the shrug should never reach the main branch. Please have in mind that you can not use any type, but the shrug emoji is included in the regex.
